SDDSDemoApp/SDDSDemoApp/Views/SegmentView/SegmentViewModel.swift

```diff
@@ -0,0 +1,203 @@
+    @Published var segmentItemAppearance: SegmentItemAppearance = SDDSSegmentItem.primary.medium.appearance {
+        didSet {
+            self.updateItems()
+        }
+    }
+    
+    // MARK: - Segment
+    @Published var layoutOrientation: SegmentLayoutOrientation = .horizontal
+    @Published var segmentSize: SegmentSize = .medium {
+        didSet {
+            self.updateSegmentAppearance()
+            self.updateSegmentItemAppearance()
+        }
+    }
+    @Published var stretch: Bool = false {
+        didSet {
+            self.updateSegmentAppearance()
+        }
+    }
+    @Published var appearance: SegmentAppearance = SDDSSegment.default.medium.appearance
+    @Published var hasBackground: Bool = true {
+        didSet {
+            self.updateSegmentAppearance()
+        }
+    }
+    @Published var variationName: String = SDDSSegmentItem.primary.name
+    @Published var isPilled: Bool = false {
+        didSet {
+            self.updateSegmentAppearance()
+            self.updateSegmentItemAppearance()
+        }
+    }
+    
+    // MARK: - Private state
+    private var segmentItemSize: SegmentItemSize = .medium
+    
+    // MARK: - Other state
+    @Published var selectedItemId: UUID?
+    
+    init() {}
+    
+    // MARK: - Private
+    private var shapeStyle: ComponentShapeStyle {
+        isPilled ? .pilled : .cornered
+    }
+    
+    private func selectSegmentItem(id: UUID) {
+        self.selectedItemId = id
+    }
+    
+    private func segmentItemSize(with segmentSize: SegmentSize) -> SegmentItemSize {
+        switch segmentSize {
+        case .large:
+            return .large
+        case .medium:
+            return .medium
+        case .small:
+            return .small
+        case .extraSmall:
+            return .extraSmall
+        }
+    }
+    
+    private func counterSize(with segmentItemSize: SegmentItemSize) -> CounterSize {
+        switch segmentItemSize {
+        case .large:
+            return .small
+        case .medium:
+            return .extraSmall
+        case .small:
+            return .extraSmall
+        case .extraSmall:
+            return .extraExtraSmall
+        }
+    }
+    
+    // MARK: - Updates
+    
+    private func updateSegmentItemSize() {
+        self.segmentItemSize = segmentItemSize(with: segmentSize)
+    }
+    
+    private func updateItems() {
+        self.updateSegmentItemSize()
+        self.updateCounterSize()
+        self.data = data.map { item in
+            let appearance = item
+                .appearance
+                .shapeStyle(shapeStyle)
+                .size(segmentItemSize)
+            
+            return .init(
+                title: item.title,
+                subtitle: item.subtitle,
+                iconAttributes: item.iconAttributes,
+                appearance: segmentItemAppearance
+                    .shapeStyle(shapeStyle)
+                    .size(segmentItemSize),
```

Comment:
```suggestion
                appearance: appearance,
```
так изменятся все цвета для существующих элементов в контейнере, а нам нужно обновлять только size и shapeStyle, потому что только size и shapeStyle из параметров контейнера могут зааффектить элементы внутри